### ISSUE TITLE
Upload to BitStore

### ISSUE THREAD

**Issue by anuveyatsu** [datopian/data-cli#16]
After publisher's credentials were authenticated, we want to upload data to BitStore.

## Acceptance criteria

* [x] function that uploads data to BitStore

## Tasks

* [x] using urls upload to BitStore
* [x] finalize

## Analysis

We want to get signed URLs for multiple files for posting data to BitStore. We need to send JWT token obtained when authenticated for DataHub. Take a look at the implementation in `dpm-py`:

Authorization and calling function that uploads data (see after this snippet):
```python
response = self._apirequest(
                method='POST',
                url='/api/datastore/authorize',
                json=file_info_for_request
            )
        filedata = response.json().get('filedata')
        if not filedata:
            raise DpmException('server did not provide upload authorization for files')

        # Upload datapackage.json
        for path in file_list:
            self._upload_file(path, filedata[path])
```

Function for uploading data:

```python
def _upload_file(self, path, data):
        '''Upload a file within the data package.'''
        # TODO: (?) echo('Uploading resource %s' % resource.local_data_path)
        local_path = join(self.datapackage.base_path, path)
        filestream = open(local_path, 'rb')

        response = requests.post(data['upload_url'],
                                 data=data['upload_query'],
                                 files={'file': filestream})

        if response.status_code not in (200, 201, 204):
            raise HTTPStatusError(
                response,
                message='Bitstore upload failed.\nError %s\n%s' % (response.status_code, response.content))
```

Finalizing:

```python
data_package_s3_url = filedata['datapackage.json']['upload_url'] + '/' +\
                              filedata['datapackage.json']['upload_query']['key']
        response = self._apirequest(
            method='POST',
            url='/api/package/upload',
            json={'datapackage': data_package_s3_url}
        )
        status = response.json().get('status', None)
        if status is None or status != 'queued':
            raise DpmException('server did not provide upload authorization for files')

        # Return published datapackage url
        return self.server + '/%s/%s' % (self.username, self.datapackage.descriptor['name'])
```

Take a look at the diagram - http://docs.datapackaged.com/developers/publish/
API endpoint: `/api/datastore/authorize`
Read more http://docs.datapackaged.com/developers/api/

**Comment by anuveyatsu** [comment]
During the implementation of uploading to BitStore, we have encountered obstacles of using `axios` library. We have been trying to replicate approach from our [Python tool][dpm-py] that uses `requests` library and uploads a file with appropriate AWS information (signatures, hashes etc.) as a FormData. However, that was not possible as we have discovered that `axios` does not support posting FormData at the moment:

- https://github.com/mzabriskie/axios/issues/789

[dpm-py]: https://github.com/frictionlessdata/dpm-py/blob/master/dpm/client/__init__.py#L199

**Comment by anuveyatsu** [comment]
FIXED in #53 
We now support pushing a Data Package from cwd.
